uaclient/cli.py

```diff
@@ -356,22 +356,18 @@ def _get_contract_token_from_cloud_identity(cfg: config.UAConfig) -> str:
         server or inability to access identity doc from metadata service.
     :raise ContractAPIError: On unexpected errors when talking to the contract
         server.
+    :raise NonAutoAttachImageError: If this cloud type does not have
+        auto-attach support.
 
     :return: contract token obtained from identity doc
     """
     cloud_type = identity.get_cloud_type()
```

Comment:
What do you think about adding a `AutoAttachCloudInstance.cloud_type` attribute, and changing `request_auto_attach_contract_token` to only take `instance`? Now that we aren't checking `cloud_type` here, it seems a little out of place.

Reply:
I like it. just pushed changes